kratos/includes/condition.h

```diff
@@ -776,68 +782,277 @@ class Condition : public GeometricalObject, public Flags
+    /** GetValuesOnIntegrationPoints methods can be used to obtain data from the condition.
+     *  They are empty on the base class, since each derived condition should know what
+     *  is expected as an output.
+     *  @param[in] rVariable The Kratos variable the data refers to.
+     *  @param[out] rValues A vector containing the value of the variable at each integration point.
+     *  @param[in] rCurrentProcessInfo ProcessInfo instance. Use it to pass global information to the condition.
+     */
+    virtual void GetValuesOnIntegrationPoints(
+        const Variable<bool>& rVariable,
+        std::vector<bool>& rValues,
+        const ProcessInfo& rCurrentProcessInfo)
+    {}
+
+    /// Get data on the integration points of the condition. Version for integer variables.
+    /** GetValuesOnIntegrationPoints methods can be used to obtain data from the condition.
+     *  They are empty on the base class, since each derived condition should know what
+     *  is expected as an output.
+     *  @param[in] rVariable The Kratos variable the data refers to.
+     *  @param[out] rValues A vector containing the value of the variable at each integration point.
+     *  @param[in] rCurrentProcessInfo ProcessInfo instance. Use it to pass global information to the condition.
+     */
+    virtual void GetValuesOnIntegrationPoints(
+        const Variable<int>& rVariable,
+        std::vector<int>& rValues,
+        const ProcessInfo& rCurrentProcessInfo)
+    {}
+
+    /// Get data on the integration points of the condition. Version for double variables.
+    /** GetValuesOnIntegrationPoints methods can be used to obtain data from the condition.
+     *  They are empty on the base class, since each derived condition should know what
+     *  is expected as an output.
+     *  @param[in] rVariable The Kratos variable the data refers to.
+     *  @param[out] rValues A vector containing the value of the variable at each integration point.
+     *  @param[in] rCurrentProcessInfo ProcessInfo instance. Use it to pass global information to the condition.
+     */
+    virtual void GetValuesOnIntegrationPoints(
+        const Variable<double>& rVariable,
+        std::vector<double>& rValues,
+        const ProcessInfo& rCurrentProcessInfo)
+    {}
+
+    /// Get data on the integration points of the condition. Version for 3-component array variables.
+    /** GetValuesOnIntegrationPoints methods can be used to obtain data from the condition.
+     *  They are empty on the base class, since each derived condition should know what
+     *  is expected as an output.
+     *  @param[in] rVariable The Kratos variable the data refers to.
+     *  @param[out] rValues A vector containing the value of the variable at each integration point.
+     *  @param[in] rCurrentProcessInfo ProcessInfo instance. Use it to pass global information to the condition.
+     */
+    virtual void GetValuesOnIntegrationPoints(
```

Comment:
we would also need a "const" version for the case in which the condition is const

Reply:
I see the point, but we need to think how to do it cleanly. Otherwise we will end up again with two methods that do different things.